sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -483,4 +527,19 @@ private FinishStatus(final boolean isFinishing, final @Nullable SpanStatus spanS
       this.spanStatus = spanStatus;
     }
   }
+
+  private static final class SpanByTimestampComparator implements Comparator<Span> {
+
+    @SuppressWarnings({"JdkObsolete", "JavaUtilDate"})
+    @Override
+    public int compare(Span o1, Span o2) {
+      final Date first =
+          Objects.requireNonNull(
+              o1.getTimestamp(), "The span timestamp should not be null after it was finished");
+      final Date second =
+          Objects.requireNonNull(
+              o2.getTimestamp(), "The span timestamp should not be null after it was finished");
+      return first.compareTo(second);
```

Comment:
That means it'll throw and we'll lose the whole transaction?

Reply:
if you mean the `requireNonNull` checks above - yes, but this should never happen, because at this point all children are finished and must have the timestamp. Maybe I misunderstood how it works though, so lemme know if you see otherwise.

Reply:
That makes sense, but I'd still apply defensive programming here, rather bailing if one of them is null than throwing an exception.